db/migrate/20241025045928_add_invites_link_to_sidebar.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+class AddInvitesLinkToSidebar < ActiveRecord::Migration[7.1]
+  def up
+    community_section_id = DB.query_single(<<~SQL).first
+      SELECT id
+      FROM sidebar_sections
+      WHERE section_type = 0
+    SQL
+
+    return if !community_section_id
+
+    max_position = DB.query_single(<<~SQL, section_id: community_section_id).first
+      SELECT MAX(ssl.position)
+        FROM sidebar_urls su
+             JOIN sidebar_section_links ssl ON su.id = ssl.linkable_id
+       WHERE ssl.linkable_type = 'SidebarUrl'
+         AND ssl.sidebar_section_id = :section_id
+         AND su.segment = 0
+    SQL
+
+    updated_rows = DB.query_hash(<<~SQL, position: max_position, section_id: community_section_id)
+      DELETE FROM sidebar_section_links
+       WHERE position > :position
+         AND sidebar_section_id = :section_id
+         AND linkable_type = 'SidebarUrl'
+      RETURNING user_id, linkable_id, linkable_type, sidebar_section_id, position + 1 AS position, created_at, updated_at
```

Comment:
Could you please explain why we need to reinsert all sidebar_section_links with position bumped by 1? I don't understand this part.

I thought that we could just insert `invite` link with `max_position + 1`

Reply:
The Invite link needs to go in the middle of the sidebar links, right after Admin and before Users (which is in the `... more` dropdown). This means that the positions of the Users link and all the links below it need to be incremented by 1 so that the new Invite link can take the old position of the Users link.

This is the sidebar links and their positions before the migration:

```
0 Topics
1 My Posts
2 Review
3 Admin
4 Users
5 About
6 FAQ
7 Groups
8 Badges
```

And this is after:

```diff
0 Topics
1 My Posts
2 Review
3 Admin
+4 Invite
5 Users
6 About
7 FAQ
8 Groups
9 Badges
```

Reply:
Ahh I see, thank you for an explanation. Technically because we have segments, even if you would put position: 10, segment: 0, it would be displayed and work correctly.

But with your migration data will be in better shape 🎊